discovery.md

```diff
@@ -0,0 +1,436 @@
+  attribute. Indicating how the `data` attribute of subscribed events will be
+  encoded.
+- Constraints:
+  - OPTIONAL
+  - If present, MUST adhere to the format specified in [RFC 2046](https://tools.ietf.org/html/rfc2046)
+
+#### dataschema
+
+- Type: `URI`
+- Description: CloudEevnts [`datacontenttype`](https://github.com/cloudevents/spec/blob/master/spec.md#dataschema)
+  attribute. This identifies the canonical storage location of the schema of
+  the `data` attribute of subscribed events.
+- Constraints:
+  - OPTIONAL
+  - If present, MUST be a non-empty URI
+
+#### dataschemacontent
+
+- Type: `String`
+- Description: An inline representation of the schema of the `data` attribute
+  encoding mechanism. This is an alternative to using the `dataschema`
+  attribute.
+- Constraints:
+  - OPTIONAL
+  - If present, MUST be a non-empty string containing a schema compatible with
+    the `datacontenttype`.
+  - If `dataschama` is present, this field MUST NOT be present.
+
+#### protocols
+
+- Type: `List` of `String`
+- Description: This field describes the different values that may be passed
+  in the `protocol` field of the CloudSubscriptions API. The protocols with
+  existing CloudEvents bindings are identified as AMQP, MQTT3, MQTT5, HTTP,
+  KAFKA, and NATS. An implementation MAY add support for further
+  protocols. All producers MUST support at least one delivery protocol, and MAY
+  support additional protocols.
+- Constraints:
+  - REQUIRED
+  - MUST be non-empty.
+- Examples:
+  - ["HTTP"]
+  - ["HTTP", "AMQP", "KAFKA"]
+
+#### extensions
+
+- Type: `Map` of `String` to `String`
+- Description: Associative map of CloudEvents [Extension Content Attributes](https://github.com/cloudevents/spec/blob/master/spec.md#extension-context-attributes)
+  that are used for this event `type`. Keys MUST be confirm to the extension
+  content attributes naming rules and value are the type of the extension
+  attribute, confirming to the CloudEvents [type system](https://github.com/cloudevents/spec/blob/master/spec.md#type-system).
+- Constraints:
+  - OPTIONAL
+
+#### subscriptionendpoint
+
+- Type: `URI-reference`
+- Description: URI indicating where CloudSubscriptions `subscribe` API calls
+  should be sent to.
+- Constraints:
+  - REQUIRED
+
+#### subscriptionconfig
+
+- Type: `Map` of `String` to `String`
+- Description: A map indicating supported options for the `config` parameter
+  for the CloudSubscriptions subscribe() API call. Keys are the name of keys
+  in the allowed config map, the values indicate the type of that parameter,
+  confirming to the CloudEvents [type system](https://github.com/cloudevents/spec/blob/master/spec.md#type-system).
+  TODO: Needs resolution with CloudSubscriptions API
+- Constraints:
+  - OPTIONAL
+- Examples:
+
+#### subscriptionttl
+
+- Type: `Integer`
+- Description: Time (in seconds) after which subscriptions of this will will
```

Comment:
s/of this will will/to this producer will/   perhaps?

Reply:
actually - going to remove this for now.